src/c_api/c_api_common.h

```diff
@@ -130,6 +134,30 @@ struct MXAPIThreadLocalEntry {
       }
     }
   }
+
+  inline static void SetupShapeArrayReturnWithBufferExInt64(
```

Comment:
If the internal implementation is the same and only differs by data type, we can create an internal template to do this?